### ISSUE TITLE
Training stuck after creating new training cache

### ISSUE THREAD

**Issue by Jessica-hub** [ultralytics/ultralytics#9303]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

Everytime I try to train the yolo model, it stuck after the new cache is created. The script is below. Anyone has anyidea why is this? I downloaded the dataset at C:\Users\olivi\YOLOv8\ultralytics-main\datasets\DOTAv1 and revised the path in the DOTAv1.yaml as "path: C:\Users\olivi\YOLOv8\ultralytics-main\datasets\DOTAv1 # dataset root dir"
. 

"""
from PIL import Image
from ultralytics import YOLO
import os

save_dir = 'C:/Users/olivi/YOLOv8/ultralytics-main/logs'
os.makedirs(save_dir, exist_ok=True)

## model arguments
data_path = r'C:\Users\olivi\YOLOv8\ultralytics-main\datasets\DOTAv1\DOTAv1.yaml' # My yaml
Project =  'C:/Users/olivi/YOLOv8/ultralytics-main/logs/' #save_dir # weight save path
experiment = 'C:/Users/olivi/YOLOv8/ultralytics-main/logs/testonobb_dotav1' #'1.yolov8_m_resume_100+300e' # experiment name
n_epochs = 5 #100
bs = 8 #8
verbose = False

model = YOLO('yolov8n-obb.pt', task = "obb")
results = model.train(data=data_path, epochs=n_epochs, batch=bs, verbose =verbose, project=Project, name=experiment)

'''

New https://pypi.org/project/ultralytics/8.1.34 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.1.29 🚀 Python-3.11.4 torch-2.0.1+cu117 CUDA:0 (NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB)
engine\trainer: task=obb, mode=train, model=yolov8n-obb.pt, data=C:\Users\olivi\YOLOv8\ultralytics-main\datasets\DOTAv1\DOTAv1.yaml, epochs=5, time=None, patience=100, batch=8, imgsz=1024, save=True, save_period=-1, cache=False, device=None, workers=8, project=C:/Users/olivi/YOLOv8/ultralytics-main/logs/, name=testonobb_dotav1, exist_ok=False, pretrained=True, optimizer=auto, verbose=False, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=C:\Users\olivi\YOLOv8\ultralytics-main\logs\testonobb_dotav1

                   from  n    params  module                                       arguments                     
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  
 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 
 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 
 22        [15, 18, 21]  1    825904  ultralytics.nn.modules.head.OBB              [15, 1, [64, 128, 256]]       
YOLOv8n-obb summary: 250 layers, 3085440 parameters, 3085424 gradients, 8.5 GFLOPs

Transferred 397/397 items from pretrained weights
TensorBoard: Start with 'tensorboard --logdir C:\Users\olivi\YOLOv8\ultralytics-main\logs\testonobb_dotav1', view at http://localhost:6006/
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks passed ✅
train: Scanning C:\Users\olivi\YOLOv8\ultralytics-main\datasets\DOTAv1\labels\train... 1411 images, 2 backgrounds, 1 corrupt: 100%|██████████| 1411/1411 [00:05<00:00, 239.06it/s]train: WARNING ⚠️ C:\Users\olivi\YOLOv8\ultralytics-main\datasets\DOTAv1\images\train\P1872.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.6534      1.8034      1.7907      1.7744      1.7483       1.723      1.6903      1.8182      1.6506      1.6087      1.6596       1.213        1.61      1.5594        1.51      1.5529      1.4456      1.4922      1.3925       1.364      1.3362       1.822      1.8426      1.8735      2.0144      2.2527
       2.097      2.1098       2.178      2.2118      2.2618      2.2772      2.2437      2.1674      2.0553      2.0337      2.0154      1.9993       1.998      1.9395      1.8671      1.8864      1.9115      1.9278      1.2735      1.1513      1.2916      1.1644      1.1396      1.3352      1.4364      1.2961
      1.1785      1.1532      1.2294      1.2884      1.3137       1.033      1.0843      1.0808      1.0927      1.0347      2.3171      2.2918      2.3637      2.4091       2.667      2.7528      2.7749      2.4934      2.4721      2.5177      2.5772        2.64       2.818      2.8019      2.8344      2.8832
      2.8976      2.8004       2.751      2.6603      2.6152      2.5574]

train: New cache created: C:\Users\olivi\YOLOv8\ultralytics-main\datasets\DOTAv1\labels\train.cache

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Jessica-hub hey there! It appears your training process gets stuck after creating a new cache. This could be due to several reasons, but let's try a few steps to troubleshoot:

1. **Check Resource Utilization:** Ensure your system has adequate resources (CPU/GPU memory) available. Training can get stuck if the system is resource-constrained.

2. **Update & Verify Installation:** Make sure you're using the latest version of the Ultralytics package and your installation is correct. You can update using:

    ```bash
    pip install -U ultralytics
    ```

    After updating, please verify your installation:

    ```python
    from ultralytics import YOLO
    print(YOLO.__version__)
    ```

3. **Corrupted Image/Label:** As mentioned in your log, there's a warning about a corrupt image/label (`P1872.jpg`) with non-normalized or out of bounds coordinates. Try removing or fixing this specific image and its corresponding label, then attempt to train again.

4. **Try Disabling Cache:** Sometimes, issues with the cache file might cause the training to halt. You can disable cache creation by adding `cache=False` in your `model.train()` call and see if that helps.

If these steps don't resolve the issue, it could be helpful to provide more details on the exact step it's getting stuck at, or any console outputs/errors you're seeing. This information would give a clearer picture of what's happening. 😊

**Comment by Jessica-hub** [comment]
Hi I tried all the method, it still does not work

**Comment by glenn-jocher** [comment]
Hey @Jessica-hub! Sorry to hear that the issue persists. Could we try one more thing? Please enable verbose mode in your training function to get more detailed logs that might pinpoint the problem:
```python
results = model.train(data=data_path, epochs=n_epochs, batch=bs, verbose=True, project=Project, name=experiment)
```
Let's see if the logs reveal anything specific when verbose is turned on. Thanks for bearing with us! 😊
